test/utils.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const generateRandomInt = (min, max) =>
+  Math.floor(Math.random() * (max - min)) + min;
+
+const generateRandomUInt8 = () => generateRandomInt(0, 2 ** 8 - 1);
+const generateRandomUInt16 = () => generateRandomInt(0, 2 ** 16 - 1);
+const generateRandomUInt32 = () => generateRandomInt(0, 2 ** 32 - 1);
```

Comment:
Math.random() generates numbers from the interval [0, 1) never resulting in 1.
That means that current implementation will never generate 2**n-1 numbers.